pkg/controller/miganalytic/miganalytics_controller.go

```diff
@@ -0,0 +1,582 @@
+	} else {
+		log.Info("Requeuing MigAnalytic, associated plan not ready.", "miganalytic", analytic.Name)
+		return reconcile.Result{RequeueAfter: time.Second * 10}, nil
+	}
+
+	// Ready
+	analytic.Status.SetReady(
+		!analytic.Status.HasBlockerCondition(),
+		ReadyMessage)
+
+	// End staging conditions.
+	analytic.Status.EndStagingConditions()
+
+	// Apply changes.
+	analytic.MarkReconciled()
+	err = r.Update(context.TODO(), analytic)
+	if err != nil {
+		log.Trace(err)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Done
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileMigAnalytic) analyze(analytic *migapi.MigAnalytic) error {
+	plan := &migapi.MigPlan{}
+
+	err := r.getAnalyticPlan(analytic, plan)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	srcCluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	client, err := srcCluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	discovery, err := discovery.NewDiscoveryClientForConfig(client.RestConfig())
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	resources, err := collectResources(discovery)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	dynamic, err := dynamic.NewForConfig(client.RestConfig())
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	analytic.Status.Analytics.Plan = plan.Name
+	log.Info("Beginning analysis of plan.", "migplan", plan.Name)
+
+	for i, namespace := range plan.Spec.Namespaces {
+		log.Info("Beginning analysis of namespace.", "namespace", namespace)
+		ns := migapi.MigAnalyticNamespace{
+			Namespace: namespace,
+		}
+
+		if analytic.Spec.AnalyzeK8SResources {
+			err := r.analyzeK8SResources(dynamic, resources, &ns, plan.Status.ExcludedResources, plan.Status.Incompatible)
+			if err != nil {
+				log.Trace(err)
+				return err
+			}
+		}
+		if analytic.Spec.AnalyzeImageCount {
+			err := r.analyzeImages(client, &ns, analytic.Spec.ListImages, analytic.Spec.MaxListImages)
+			if err != nil {
+				log.Trace(err)
+				return err
+			}
+		}
+		if analytic.Spec.AnalyzePVCapacity {
+			err := r.analyzePVCapacity(client, &ns)
+			if err != nil {
+				log.Trace(err)
+				return err
+			}
+		}
+
+		analytic.Status.Analytics.Namespaces = append(analytic.Status.Analytics.Namespaces, ns)
+		analytic.Status.Analytics.K8SResourceTotal += ns.K8SResourceTotal
+		analytic.Status.Analytics.ExcludedK8SResourceTotal += ns.ExcludedK8SResourceTotal
+		analytic.Status.Analytics.IncompatibleK8SResourceTotal += ns.IncompatibleK8SResourceTotal
+		analytic.Status.Analytics.ImageCount += ns.ImageCount
+		analytic.Status.Analytics.ImageSizeTotal.Add(ns.ImageSizeTotal)
+		analytic.Status.Analytics.PVCapacity.Add(ns.PVCapacity)
+		analytic.Status.Analytics.PVCount += ns.PVCount
+		analytic.Status.Analytics.PercentComplete = (i + 1) * 100 / len(plan.Spec.Namespaces)
+
+		err = r.Update(context.TODO(), analytic)
+		if err != nil {
+			log.Trace(err)
+			return err
```

Comment:
Anywhere in this PR outside of 
```
func (r *ReconcileMigAnalytic) Reconcile(request reconcile.Request) (reconcile.Result, error)
```
We should be returning _all_ errors using with liberr.Wrap(). As an example:

```
# Replace this...
if err != nil {
   log.Trace(err)
   return err
}

# With this:
if err != nil {
   return liberr.Wrap(err)
}
```

Only in the root `Reconcile()` methods should we still be doing log.Trace()

More info: https://github.com/konveyor/mig-controller/pull/595